### DISCUSSION TITLE
Why is Google Bot tracking dynamic routes?

### DISCUSSION THREAD

**Discussion by rserafim** [vercel/next.js#36911] [Help]
![Untitled-1](https://user-images.githubusercontent.com/48859044/168426682-04ba0048-5565-4a4c-9d37-31fa217a859d.png)

**Comment by rserafim** [comment]
![image](https://user-images.githubusercontent.com/48859044/168702641-60e192df-e9da-49c9-837e-97381b75e029.png)

**Reply by icyJoseph** [reply]
Have you somehow told the bot to look for these? Perhaps a site-map or something like that? or if there any link going to these pages? Weird indeed.

**Comment by rserafim** [comment]
Worse than not! Sitemap is generated automatically. I've searched through all the code and haven't found anything like []

If I block these urls in robots.txt would solve it?

Disallow: /[slug_city]/empresas-na-cidade
Disallow: /empresas/[slug_city]/[category]
Disallow: /[slug_city]/categorias/[category_father]
Disallow: /[slug_city]/[neighborhood]/[category]/[slug]/[id]

**Reply by icyJoseph** [reply]
Doesn't feel like a good solution though. I have another project that uses one dynamic route, and I see none of this in the Google Search Console. 

I guess you'd still have to do it, to make the bot stop.

Would it be too much to ask for a repository to look at? I am curious about what happened here.

**Reply by rserafim** [reply]
I also have another smaller project and I never noticed these hits.

Unfortunately I cannot make it available for confidentiality reasons.

And I can't get support from Vercel either. They haven't responded to me in weeks. =/

**Comment by mahali00** [comment]
@rserafim To prevent dynamic URLs from being crawled and causing server errors in Google Search Console, you can use the `getStaticPaths` function and specify all routes. Additionally, set `fallback` to `false` so that any paths not specified in `getStaticPaths` will return a `404`  page.

**Reply by rserafim** [reply]
Unable to do this on a large project. That's not the solution.

**Comment by rserafim** [comment]
The problem is not in the sitemap. Note that the crawl did not come from any sitemap, but from a referring page.

![code](https://user-images.githubusercontent.com/48859044/214291432-43e8a6a7-22e5-4901-916c-e0dfd206e788.png)

From what I understand the error is in the json that NextJS is generating. See that he inserts the dynamic path and with that the BOT accesses it.
![imgcheck](https://user-images.githubusercontent.com/48859044/214293530-144dd9ca-c5c8-4da6-b77f-8cedd6656705.jpg)

![image](https://user-images.githubusercontent.com/48859044/214292513-58b14cad-9dd1-4d6e-a8b2-918df9d92cb3.png)

**Comment by d-vorobyov** [comment]
In our case `<Link>` on `entity/[slug]` pages with `href=""` generated wrong <a> tag, and crawler followed it
<img width="450" alt="image" src="https://github.com/vercel/next.js/assets/78736549/b836afd8-dafb-4a72-a7c8-c9d5ba21f6ec">
